### PR TITLE
Fix data-astro-raw logic

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Closes https://github.com/withastro/compiler/issues/242 and https://github.com/withastro/astro/issues/2346
- Fixes logic error in `data-astro-raw` detection

## Testing

Multiple tests added

## Docs

Bug fix only

### CODE REVIEW

File: internal/token.go [hunk 1]
```diff
@@ -931,18 +931,12 @@ loop:
...
-			c := key[i]
-			if c != s[i] {
-				continue loop
-			}
+		if string(key) == s {
+			return true
 		}
-		return true
 	}
 	return false
```
Comment (natemoo-re) [inline internal/token.go:941]: This was the fix! `data-astro-raw` was previously matching any slice (from 0 to len) of `data-astro-raw`, so `data` would incorrectly be matched.

The fix was to simplify the loop logic and just do simple string comparison.

Comment (matthewp) [reply internal/token.go:941]: oops

**Comment by matthewp** [comment]
Great catch! lgtm

File: internal/token.go (hunk 1 above)
Comment (Morritz) [reply internal/token.go:941]: > This was the fix! `data-astro-raw` was previously matching any slice (from 0 to len) of `data-astro-raw`, so `data` would incorrectly be matched.
> 
> The fix was to simplify the loop logic and just do simple string comparison.
I didnt expect it to be nested this deep, I didnt even knew that the golang was used for the compiler

**Pull request merged**